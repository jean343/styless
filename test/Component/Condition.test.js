import React from 'react';
import renderer from 'react-test-renderer';
import styled from 'styled-components';
import 'jest-styled-components';

test('Test and', () => {
    const Div = styled.div`
	    opacity1: if((@errorText) and (@warning), 1, 0);
	    opacity2: if(@errorText and @warning, 1, 0);
	`;
    expect(renderer.create(<Div/>).toJSON()).toMatchSnapshot();
    expect(renderer.create(<Div errorText/>).toJSON()).toMatchSnapshot();
    expect(renderer.create(<Div warning/>).toJSON()).toMatchSnapshot();
    expect(renderer.create(<Div errorText warning/>).toJSON()).toMatchSnapshot();
});

test('Test or', () => {
    const Div = styled.div`
	    opacity1: if((@errorText) or (@warning), 1, 0);
	    opacity2: if(@errorText or @warning, 1, 0);
	`;
    expect(renderer.create(<Div/>).toJSON()).toMatchSnapshot();
    expect(renderer.create(<Div errorText/>).toJSON()).toMatchSnapshot();
    expect(renderer.create(<Div warning/>).toJSON()).toMatchSnapshot();
    expect(renderer.create(<Div errorText warning/>).toJSON()).toMatchSnapshot();
});

test('Test =', () => {
    const Div = styled.div`
	    opacity1: if(@a = @b, 1, 0);
	`;
    expect(renderer.create(<Div a={1} b={1}/>).toJSON()).toMatchSnapshot();
    expect(renderer.create(<Div a={1} b={0}/>).toJSON()).toMatchSnapshot();
    expect(renderer.create(<Div a="lala" b="lala"/>).toJSON()).toMatchSnapshot();
    expect(renderer.create(<Div a="lala" b={false}/>).toJSON()).toMatchSnapshot();
});

test('Test greater than', () => {
    const Div = styled.div`
	    opacity: if(luma(@bg) > 50%, 1, 0);
	    opacity: if(luma(@bg) >= 50%, 1, 0);
	    opacity: if(luma(@bg) < 50%, 1, 0);
	    opacity: if(luma(@bg) <= 50%, 1, 0);
	`;
    expect(renderer.create(<Div bg="black"/>).toJSON()).toMatchSnapshot();
    expect(renderer.create(<Div bg="white"/>).toJSON()).toMatchSnapshot();
});