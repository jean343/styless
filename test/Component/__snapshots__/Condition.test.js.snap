// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test = 1`] = `
.c0 {
  opacity1: 1;
}

<div
  className="c0"
/>
`;

exports[`Test = 2`] = `
.c0 {
  opacity1: 0;
}

<div
  className="c0"
/>
`;

exports[`Test = 3`] = `
.c0 {
  opacity1: 1;
}

<div
  className="c0"
/>
`;

exports[`Test = 4`] = `
.c0 {
  opacity1: 0;
}

<div
  className="c0"
/>
`;

exports[`Test and 1`] = `
.c0 {
  opacity1: 0;
  opacity2: 0;
}

<div
  className="c0"
/>
`;

exports[`Test and 2`] = `
.c0 {
  opacity1: 0;
  opacity2: 0;
}

<div
  className="c0"
/>
`;

exports[`Test and 3`] = `
.c0 {
  opacity1: 0;
  opacity2: 0;
}

<div
  className="c0"
/>
`;

exports[`Test and 4`] = `
.c0 {
  opacity1: 1;
  opacity2: 1;
}

<div
  className="c0"
/>
`;

exports[`Test greater than 1`] = `
.c0 {
  opacity: 0;
  opacity: 0;
  opacity: 1;
  opacity: 1;
}

<div
  className="c0"
/>
`;

exports[`Test greater than 2`] = `
.c0 {
  opacity: 1;
  opacity: 1;
  opacity: 0;
  opacity: 0;
}

<div
  className="c0"
/>
`;

exports[`Test or 1`] = `
.c0 {
  opacity1: 0;
  opacity2: 0;
}

<div
  className="c0"
/>
`;

exports[`Test or 2`] = `
.c0 {
  opacity1: 1;
  opacity2: 1;
}

<div
  className="c0"
/>
`;

exports[`Test or 3`] = `
.c0 {
  opacity1: 1;
  opacity2: 1;
}

<div
  className="c0"
/>
`;

exports[`Test or 4`] = `
.c0 {
  opacity1: 1;
  opacity2: 1;
}

<div
  className="c0"
/>
`;
